### ISSUE TITLE
Arduino Seems to Enter Panic 3 Mode on Startup

### ISSUE THREAD

**Issue by DHolden24** [VinDuv/switch-arduino-controller#8]
I'm trying to get this working with my Switch.

I was able to successfully run the make command and flash the hex files to my Arduino (or at least, I have no indication that it was unsuccessful).
However, when I connect it to my Switch, it seems to immediately enter the panic mode 3 (TX and RX lights are off, L light blinks 3 times and then is off for a longer gap before repeating).

I went through an issue that seems similar (https://github.com/VinDuv/switch-arduino-controller/issues/4) to try and follow some of the troubleshooting steps, but nothing there has seemed to work.

Is there anything else I can try to get this running?

**Comment by VinDuv** [comment]
panic(3) on the main microcontroller means that it’s not able to communicate with the USB interface microcontroller.
I’ve pushed a branch [test-tools](https://github.com/VinDuv/switch-arduino-controller/tree/test-tools) with a test tool for the USB interface.
Could you check out this branch, go to the `src/test-tools` directory, and run `make flash-test-usb-repeat-1sec` while the programmer is connected to the USB interface ICSP? This will flash a program to the USB interface that press D-pad left then right once per second, without needing the main microcontroller.
If this works once plugged in a Switch, it means the serial interface between the microcontrollers has an issue. If it doesn’t, there is probably something wrong with the USB interface microcontroller.

**Comment by DHolden24** [comment]
It doesn't seem to work, no button presses seem to occur. I've attached the make command output, it gives "Error 1" at the end, but I don't seen anything that looks wrong to me.

```make -f Makefile.lufa TARGET=test-usb-repeat-1sec TEST_SRC="test-usb-repeat-1sec.c test-usb-repeat.c test-usb-base.c"
make[1]: Entering directory '/home/dholden/git/switch-arduino-controller/src/test-tools'
 [INFO]    : Begin compilation of project "test-usb-repeat-1sec"...

avr-gcc (AVR_8_bit_GNU_Toolchain_3.7.0_1796) 7.3.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

 [LNK]     : Linking object files into "test-usb-repeat-1sec.elf"
avr-gcc obj/test-usb-repeat-1sec.o obj/test-usb-repeat.o obj/test-usb-base.o obj/usb-descriptors.o obj/HIDParser.o obj/Device_AVR8.o obj/EndpointStream_AVR8.o obj/Endpoint_AVR8.o obj/Host_AVR8.o obj/PipeStream_AVR8.o obj/Pipe_AVR8.o obj/USBController_AVR8.o obj/USBInterrupt_AVR8.o obj/ConfigDescriptors.o obj/DeviceStandardReq.o obj/Events.o obj/HostStandardReq.o obj/USBTask.o -o test-usb-repeat-1sec.elf -lm -Wl,-Map=test-usb-repeat-1sec.map,--cref -Wl,--gc-sections -Wl,--relax -mmcu=atmega16u2
 [OBJCPY]  : Extracting HEX file data from "test-usb-repeat-1sec.elf"
avr-objcopy -O ihex -R .eeprom -R .fuse -R .lock -R .signature test-usb-repeat-1sec.elf test-usb-repeat-1sec.hex
 [OBJCPY]  : Extracting EEP file data from "test-usb-repeat-1sec.elf"
avr-objcopy -O ihex -j .eeprom --set-section-flags=.eeprom="alloc,load" --change-section-lma .eeprom=0 --no-change-warnings test-usb-repeat-1sec.elf test-usb-repeat-1sec.eep || exit 0
 [OBJCPY]  : Extracting BIN file data from "test-usb-repeat-1sec.elf"
avr-objcopy -O binary -R .eeprom -R .fuse -R .lock -R .signature test-usb-repeat-1sec.elf test-usb-repeat-1sec.bin
 [OBJDMP]  : Extracting LSS file data from "test-usb-repeat-1sec.elf"
avr-objdump -h -d -S -z test-usb-repeat-1sec.elf > test-usb-repeat-1sec.lss
 [NM]      : Extracting SYM file data from "test-usb-repeat-1sec.elf"
avr-nm -n test-usb-repeat-1sec.elf > test-usb-repeat-1sec.sym
 [SIZE]    : Determining size of "test-usb-repeat-1sec.elf"

avr-size --mcu=atmega16u2 --format=avr test-usb-repeat-1sec.elf
AVR Memory Usage
----------------
Device: atmega16u2

Program:    3248 bytes (19.8% Full)
(.text + .data + .bootloader)

Data:         69 bytes (13.5% Full)
(.data + .bss + .noinit)


 [INFO]    : Finished building project "test-usb-repeat-1sec".
make[1]: Leaving directory '/home/dholden/git/switch-arduino-controller/src/test-tools'
rm -f *.bin *.elf *.eep *.lss *.map *.sym
avrdude -p m16u2 -c avrispmkii -P usb -U flash:w:test-usb-repeat-1sec.hex -U lfuse:w:0xFF:m -U hfuse:w:0xD9:m -U efuse:w:0xF4:m -U lock:w:0x0F:m

avrdude done.  Thank you.

make: *** [Makefile:10: flash-test-usb-repeat-1sec] Error 1```

**Comment by VinDuv** [comment]
Hmm, it’s weird that avrdude doesn’t say anything, but I’m pretty sure it’s failing. (It says “avrdude done.  Thank you.” even when it fails due to the programmer not being attached…)
Try to run the flash command manually, to see if it produces more output:
```
avrdude -p m16u2 -c avrispmkii -P usb -U flash:w:test-usb-repeat-1sec.hex -U lfuse:w:0xFF:m -U hfuse:w:0xD9:m -U efuse:w:0xF4:m -U lock:w:0x0F:m
```
You can also check your avrdude version by running `avrdude -h` (last line of output); I’ve tested versions 7.1 (on Linux) and 7.3 (on macOS) without issues.

**Comment by DHolden24** [comment]
I got it working, thanks for your help!

Running the flash command did give more context. Turns out the main issue seemed to be that I needed to change the programmer that avrdude was using (I need `usbasp`).

I also had an older version of avrdude, I updated it, but I'm not sure if that was contributing or not.

This is my first time using an Arduino, so sorry if that was a basic mistake. Thanks for your help!
